engine/src/main/java/org/destinationsol/world/generators/SolarSystemGenerator.java

```diff
@@ -349,6 +356,8 @@ protected void initializeRandomBeltGenerators(float beltChance) {
                 try {
                     if (SolRandom.seededTest(beltChance)) {
                         FeatureGenerator newFeatureGenerator = newFeatureGeneratorType.newInstance();
```

Comment:
Why are you declaring it a `FeatureGenerator` instance if it is a `BeltGenerator`? If you're going to cast it, you should probably either do that here or at the beginning of the previous `if` (before the line `Class<? ...`)

Reply:
Good point, I'll address that.